### ISSUE TITLE
Image tag specifier is not used when pushing through build goal

### ISSUE THREAD

**Issue by nistvan86** [spotify/docker-maven-plugin#40]
When building an image with a tag using the build goal, and pushImage is set to true, push refers to the tag-less image which results in pushing the latest tag only.

This doesn't work:
```
      <plugin>
        <groupId>com.spotify</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <version>0.0.21</version>
        <executions>
          <execution>
            <id>build</id>
            <phase>package</phase>
            <goals>
              <goal>build</goal>
            </goals>
            <configuration>
              <dockerHost>tcp://docker.host:5555</dockerHost>
              <imageName>docker.registry/image:withtag</imageName>
              <dockerDirectory>${basedir}/src/main/docker</dockerDirectory>
              <pushImage>true</pushImage>
              <resources>
                <resource>
                  <targetPath>.</targetPath>
                  <directory>${project.build.directory}</directory>
                  <include>docker-assembly.tar</include>
                </resource>
              </resources>
            </configuration>
          </execution>
          <execution>
            <id>push</id>
            <phase>package</phase>
            <goals>
              <goal>push</goal>
            </goals>
            <configuration>
              <dockerHost>tcp://docker.host:5555</dockerHost>
              <imageName>docker.registry/image:withtag</imageName>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

By separating the build and push executions it works though:
```
      <plugin>
        <groupId>com.spotify</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <version>0.0.21</version>
        <executions>
          <execution>
            <id>build</id>
            <phase>package</phase>
            <goals>
              <goal>build</goal>
            </goals>
            <configuration>
              <dockerHost>tcp://docker.host:5555</dockerHost>
              <imageName>docker.registry/image:withtag</imageName>
              <dockerDirectory>${basedir}/src/main/docker</dockerDirectory>
              <resources>
                <resource>
                  <targetPath>.</targetPath>
                  <directory>${project.build.directory}</directory>
                  <include>docker-assembly.tar</include>
                </resource>
              </resources>
            </configuration>
          </execution>
          <execution>
            <id>push</id>
            <phase>package</phase>
            <goals>
              <goal>push</goal>
            </goals>
            <configuration>
              <dockerHost>tcp://docker.host:5555</dockerHost>
              <imageName>docker.registry/image:withtag</imageName>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by davidxia** [comment]
@nistvan86 I just tried with 0.2.13 and the image tag is used during the push. Your first pom example should work now with more recent versions. If it does, can we close this?

**Comment by adowner** [comment]
I have created a pull request which adds the ability to only push images explicitly defined in the tags property. This have been useful to us when we have generic build machines which build a lot of tags on a specific image. we don't want to push all the images under imageName all the time. only the ones that have been been built by a specific process. This would seem to apply to the OP's scenario.

**Comment by mattnworb** [comment]
AFAIK this was resolved in #127. @nistvan86 please feel free to re-open if that does not fix your issue.
